### PR TITLE
restructure post editing for AJAX loading of resources

### PR DESCRIPTION
**Pull request by edraut**
added processing and processing failed statuses for versions
added views for processing and processing failed versions
added foreign office browser push for processing version clone progress meter

### CODE REVIEW

File: app/assets/javascripts/pulitzer.js [hunk 1]
```diff
@@ -54,6 +55,14 @@ var ContentElementEditor = Class.extend({
   }
 })
 
+$(document).ajaxComplete(function(){
+  $.each(window.any_time_manager.recordedObjects["RichTextEditor"], function(){
+    if(this.$form.parents('body').length == 0){ //the form has been removed from the dom
+      this.editor.fire('destroy:composer')
```
Comment (gomayonqui) [inline app/assets/javascripts/pulitzer.js]: :+1:

File: app/views/pulitzer/content_elements/_new.html.erb [hunk 2]
```diff
@@ -1,17 +1,17 @@
-<%= form_for content_element, html: ajax_form_hash("#version_content_element_container_#{dom_id(version)}", insert_method: 'append', sub_class: "AddALineForm", container: "#new_version_content_element_#{dom_id(version)}") do |f| %>
+<%= form_for content_element, html: ajax_form_hash(dom_target(version, :free_form_content_element_container), insert_method: 'append', sub_class: "AddALineForm", container: dom_target(version, :new_free_form_content_element)) do |f| %>
   <%= f.collection_select :content_element_type_id, Pulitzer::ContentElementType.all, :id, :name, {},
-  { data: { "revealer" => true, "sub-type" => "FormFieldRevealer", "revealer-children-id" => dom_id(version, :revealer) }} %>
+  revealer(dom_id(version, :content_element_type_revealer), highlander: true) %>
 
-  <div data-revealer-target="<%= dom_id(version, :revealer) %>"></div>
+  <div <%= revealer_target_attrs dom_id(version, :content_element_type_revealer) %>></div>
```
Comment (gomayonqui) [inline app/views/pulitzer/content_elements/_new.html.erb]: So cool that you are updating the revealers with the new helpers

File: app/views/pulitzer/versions/_edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    <ul>
+      <% version.cloning_errors.each do |error| %>
+        <li><%= error %></li>
+      <% end %>
+    </ul>
+  </div>
+<% elsif version.processing? %>
+  <h2> Processing Preview for <%= version.title %> </h2>
+  <div class="pulitzer-row">
+    <progress value="0" max="<%= version.post.active_version.content_elements.count + version.post.active_version.post_tags.count + 2 %>" <%= listener_attrs version, :processed_element_count if defined? ForeignOffice %>></progress>
```
Comment (gomayonqui) [inline app/views/pulitzer/versions/_edit.html.erb]: Why + 2 to the post_tags count?

**Comment by gomayonqui** [comment]
Great job @edraut :cake:

File: app/views/pulitzer/versions/_edit.html.erb (hunk 3 above)
Comment (edraut) [inline app/views/pulitzer/versions/_edit.html.erb]: It's actually a bit silly, I just wanted to count as many things as possible to keep the progress meter moving smoothly. So I count each content element as it's copied, then I count each post tag as it's copied, then I count once for updating the status in the database, and once more for generating the url for the new preview version.

**Pull request merged**